src/routing/core.rs

```diff
@@ -1127,9 +1127,38 @@ impl Core {
             return Ok(vec![]);
         }
 
+        let old_adults: BTreeSet<_> = self
+            .section
+            .active_members()
+            .filter_map(|peer| {
+                if self.section.is_elder(peer.name()) {
+                    None
+                } else {
+                    Some(*peer.name())
+                }
+            })
+            .collect();
+
         let snapshot = self.state_snapshot();
-        self.section.merge(section)?;
+        self.section.merge(section.clone())?;
```

Comment:
Would be nice to have a log here saying we've updated our knowledge of our section.